packages/snap-controller/src/Recommendation/RecommendationController.ts

```diff
@@ -73,6 +74,21 @@ export class RecommendationController extends AbstractController {
 		this.config = deepmerge(defaultConfig, this.config);
 		this.store.setConfig(this.config);
 
+		this.eventManager.on('afterStore', async (search: AfterStoreObj, next: Next): Promise<void | boolean> => {
+			await next();
+			const controller = search.controller as RecommendationController;
+			if (controller.store.loaded && !controller.store.error) {
+				if (controller.store.results.length === 0) {
+					this.track.product.render();
+				}
+				controller.store.results.forEach((result) => {
+					if (result.type === 'product') {
+						this.track.product.render(result as Product);
+					}
+				});
+			}
+		});
+
```

Comment:
Should use similar technique as was refactored in SearchController - build out event from store data and handle no results case.